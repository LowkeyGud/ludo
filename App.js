import React from 'react';
import {Text} from 'react-native';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';
import Navigation from './src/navigation/Navigation';
import {persistor, store} from './src/redux/store';

const App = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={<Text>Loadin..</Text>} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};

export default App;
