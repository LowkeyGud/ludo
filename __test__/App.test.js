import {render} from '@testing-library/react-native';
import React from 'react';
import App from '../App';

jest.useFakeTimers();

// Mock redux-persist
jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((_config, reducers) => reducers),
    // Mock other exports if needed
  };
});

test('The Main App Renders Correctly', () => {
  const {toJSON} = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});
