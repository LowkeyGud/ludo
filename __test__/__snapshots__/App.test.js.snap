// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Main App Renders Correctly 1`] = `null`;
